src/track/globaltrackcache.cpp

```diff
@@ -510,25 +545,19 @@ void GlobalTrackCache::resolve(
                 << "Cache miss - allocating track"
                 << trackRef;
     }
-    auto deletingPtr = std::unique_ptr<Track, void (&)(Track*)>(
+    auto deletingPtr = std::unique_ptr<Track, GlobalTrackCacheEntry::TrackDeleter>(
             new Track(
                     std::move(fileInfo),
                     std::move(pSecurityToken),
                     std::move(trackId)),
-            deleteTrack);
-    // Track objects live together with the cache on the main thread
-    // and will be deleted later within the event loop. But this
-    // function might be called from any thread, even from worker
-    // threads without an event loop. We need to move the newly
-    // created object to the target thread.
-    deletingPtr->moveToThread(QApplication::instance()->thread());
+            GlobalTrackCacheEntry::TrackDeleter(m_deleteTrack));
```

Comment:
I don't understand why we need a functor that has the same interface as deleteTrackFn.
Normally a functor is used as an adapter to adjust the interface or carry additional objects. 
cant we just pass the original deleteTrackFn to simplify the code?

Reply:
Wrapping a custom function pointer requires state. Otherwise we would not able to add common code before or after calling the function pointer.

Reply:
Ah OK, I missed that.